pkg/engine/lifecycletest/refresh_test.go

```diff
@@ -304,6 +304,122 @@ func TestRefreshDeleteDependencies(t *testing.T) {
 	}
 }
 
+func TestRefreshDeletePropertyDependencies(t *testing.T) {
+	t.Parallel()
+
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				ReadF: func(
+					urn resource.URN, id resource.ID, inputs, state resource.PropertyMap,
+				) (plugin.ReadResult, resource.Status, error) {
+					if urn.Name() == "resA" {
+						return plugin.ReadResult{}, resource.StatusOK, nil
+					}
+
+					return plugin.ReadResult{Outputs: resource.PropertyMap{}}, resource.StatusOK, nil
+				},
+			}, nil
+		}),
+	}
+
+	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
+		resA, err := monitor.RegisterResource("pkgA:m:typA", "resA", true)
+		assert.NoError(t, err)
+
+		_, err = monitor.RegisterResource("pkgA:m:typA", "resB", true, deploytest.ResourceOptions{
+			PropertyDeps: map[resource.PropertyKey][]resource.URN{
+				"propB1": {resA.URN},
+			},
+		})
+
+		assert.NoError(t, err)
+		return nil
+	})
+
+	hostF := deploytest.NewPluginHostF(nil, nil, programF, loaders...)
+
+	p := &TestPlan{
+		Options: TestUpdateOptions{HostF: hostF, UpdateOptions: UpdateOptions{Parallel: 4}},
```

Comment:
Is the parallelism factor here significant?  I'd just leave it out if it isn't, not to confuse future readers.